src/update.c

```diff
@@ -314,7 +412,11 @@ int main_update()
 	/* Step 3: setup manifests */
 
 	/* get the from/to MoM manifests */
-	current_manifest = load_mom(current_version, false);
+	if (system_on_mix()) {
+		current_manifest = load_mom(current_version, false, mix_exists);
```

Comment:
How would you get into a state where you system is on a mix (`system_on_mix()` - `/usr/share/defaults/swupd/mixed` exists) but a mix doesn't exist (`/usr/share/mix` doesn't have any bundles in it)?

Can you be on an empty mix? If not, this could just be

`current_manifest = load_mom(current_version, false, system_on_mix());`

Reply:
You can technically have an empty mix with no bundles added, and you just want to have moved your system to the mix state, or you simply didn't add anything yet and are just incrementing the version/upstream content only.

Reply:
I see, thanks for explaining.